source/_posts/2019-01-14-Google-Home.md

```diff
@@ -0,0 +1,55 @@
+---
+title: A Guide to Google Home Integrations
+date: 2019-01-14 10:00:00
+tags: server,google,home,ai,api,cloud,serverless
+authorIds: 
+- brto
+categories:
+- Vapor
+---
+
+Within the trend of IoT Google Home is amongst the most popular. Thankfully, Google has made the development of actions quite approachable. To get an overview of the process, let's create a weather integration together.
+
+## Actions on Google
+Creating your Google Home integration starts at Actions on Google (AoG). Quoting Google: 
+> Actions on Google is the platform for developers to extend the Google Assistant. Join this emerging ecosystem by developing actions to engage users on Google Home, Pixel, and many other surfaces where the Google Assistant will be available.
+
+
+Now, it's time to create your action. Under the _Build_ section, click _ADD ACTION_ and finally select _Custom intent_. After creating the action, Google will redirect you to their [Dialogflow](http://dialogflow.com) service.
+
+## Dialogflow
+Dialogflow is Google's intuitive natural language processing (NLP) tool. Here is where you train the AI to recognize intents. This is by far the smoothest part of creating a Google Home integration. Create an intent and just type in a few example phrases to train the AI.
+
+- How's the weather today?
+- What's the weather?
+- How's the weather in `Copenhagen`?
+
+After configuring the training phrases, make sure to check the box _Enable webhook call for this intent_. Doing so allows you to add fulfillments programmatically.
+
+### Fulfillment
+A fulfillment is a response to an intent. Programmatic fulfillments allow you to do many things: present a card with information, prompt a user for more information or show an error. Fulfillments use the webhook REST API and can be severed through cloud functions or self hosting.
+
+#### Cloud Functions
+In the _Fulfillment_ tab on Dialogflow you have the option to edit a fulfillment and deploy it to Google Cloud Functions all within the inline editor. If you're short on time or your project isn't that large, this is by far the simpler option of the two. Using NodesJS and the 
+
+#### Self-hosting
+If you're working on a larger project or cloud functions aren't your thing, you can host your server and configure Dialogflow to fulfill all requests using your server. As most major programming languages have a fulfillment library, you have a much wider choice in what to develop your fulfillment in.
```

Comment:
Suggestion: `you can host your own server and configure Dialogflow to fulfill all requests using that one instead.`